### PR TITLE
Webpack: Allow watching workspaces

### PR DESCRIPTION
**Pull request by jackw**
**What is this feature?**

Fixes an issue with webpack ignoring symlinks which yarn node-linker uses for workspaces when set to node-modules.

**Why do we need this feature?**

To have the same dev experience as before switching away from yarn pnp

**Who is this feature for?**

Grafana contributors/developers

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: scripts/webpack/webpack.common.js [hunk 1]
```diff
@@ -42,7 +42,6 @@ module.exports = {
       https: false,
       string_decoder: false,
     },
-    symlinks: false,
```
Comment (joshhunt) [inline scripts/webpack/webpack.common.js:45]: This is the main fix for watching packages, right?

**Review by joshhunt — APPROVED** [review]
pushed up some comments, looks good to go, thanks for fixing this quickly!

File: scripts/webpack/webpack.common.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ module.exports = {
         '@locker/near-membrane-dom/custom-devtools-formatter.js'
       ),
     },
-    modules: ['node_modules', path.resolve('public')],
+    modules: ['node_modules', path.resolve('node_modules'), path.resolve('public')],
```
[the remark this answers is not in the document]
Comment (jackw) [reply scripts/webpack/webpack.common.js]: Exactly. The enterprise files cannot resolve packages without this.

File: scripts/webpack/webpack.common.js (hunk 1 above)
Comment (jackw) [reply scripts/webpack/webpack.common.js:45]: tldr; yes. 

By default files are watched with hashes/timestamps however inside node_modules directory package name and version is used for watching _unless_ it's a symlink which is required for `npm link` or workspaces (outside of yarn PnP) to function.

**Pull request merged**